validator/src/main/java/com/google/daq/mqtt/registrar/Registrar.java

```diff
@@ -508,13 +509,24 @@ private void shutdown() {
     }
   }
 
-  private Map<String, LocalDevice> loadLocalDevices() {
+  private Map<String, LocalDevice> loadLocalDevices(Set<String> specifiedDevices) {
     Preconditions.checkNotNull(siteDir, "site directory");
     File devicesDir = new File(siteDir, DEVICES_DIR);
     if (!devicesDir.isDirectory()) {
       throw new RuntimeException("Not a valid directory: " + devicesDir.getAbsolutePath());
     }
-    String[] devices = devicesDir.list();
+
+    String[] devices;
```

Comment:
declare final (just add the "final" keywoard before String[]) -- this just means that it should be assigned once, and exactly once, and then is just enforced by the compiler.

Reply:
Done